### PR TITLE
Respond to configuration changes to enable/disable the language server

### LINKED ISSUE #349: Enable the language server by default.
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

The MDX language server is now based on Volar and is pretty stable. It’s time to unmark it as experimental and enable it by default.

I still want it to be possible for users to disable the language server in case they run into issues.

### Solution

Let’s investigate if Volar has builtin functionality for this. Otherwise provide it upstream if desirable, or it in `vscode-mdx`.

### Alternatives

Don’t provide the possibility to disable the language server.

### PR DESCRIPTION
**Pull request by johnsoncodehk**
Closes #349

### CODE REVIEW

File: packages/vscode-mdx/package.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
         "properties": {
           "mdx.experimentalLanguageServer": {
```
Comment (remcohaszing) [inline packages/vscode-mdx/package.json:67]: I would like to rename this property to mark this as no longer experimental. I’m currently thinking of `mdx.server.enable`, but I’m open to suggestions. (`<feature>.enable` is a quite common unofficial convention for VSCode settings.)

```suggestion
          "mdx.server.enable": {
```

File: packages/vscode-mdx/README.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ You can install this extension from the [Marketplace](https://marketplace.visual
 This extension provides the following settings:
 
 *   `mdx.experimentalLanguageServer`: Enable experimental IntelliSense support
```
Comment (remcohaszing) [inline packages/vscode-mdx/README.md:19]:
```suggestion
*   `mdx.server.enable`: Enable the MDX language server. (`boolean`, default: true)
```

File: packages/vscode-mdx/src/extension.js [hunk 3]
```diff
@@ -45,15 +52,14 @@ export async function activate(context) {
...
-  activateAutoInsertion([client], (document) => document.languageId === 'mdx')
+  tryRestartServer()
 
   context.subscriptions.push(
-    languages.registerDocumentDropEditProvider(
-      {language: 'mdx'},
-      documentDropEditProvider
-    )
+    workspace.onDidChangeConfiguration((event) => {
+      if (event.affectsConfiguration('mdx.experimentalLanguageServer')) {
```
Comment (remcohaszing) [inline packages/vscode-mdx/src/extension.js:59]:
```suggestion
      if (event.affectsConfiguration('mdx.server.enable')) {
```

File: packages/vscode-mdx/README.md [hunk 4]
```diff
@@ -17,7 +17,7 @@ You can install this extension from the [Marketplace](https://marketplace.visual
...
-    for MDX files.  (`boolean`, default: false)
+    for MDX files.  (`boolean`, default: true)
```
Comment (remcohaszing) [inline packages/vscode-mdx/README.md:20]:
```suggestion
```

File: packages/vscode-mdx/src/extension.js [hunk 5]
```diff
@@ -63,13 +69,51 @@ export async function activate(context) {
       languageServerProtocol
     }
   }
+
+  async function tryRestartServer() {
+    await stopServer()
+    if (workspace.getConfiguration('mdx').get('experimentalLanguageServer')) {
```
Comment (remcohaszing) [inline packages/vscode-mdx/src/extension.js:75]:
```suggestion
    if (workspace.getConfiguration('mdx').get('server.enable')) {
```

**Review by remcohaszing — CHANGES_REQUESTED** [review]

File: packages/vscode-mdx/package.json (hunk 1 above)
Comment (johnsoncodehk) [reply packages/vscode-mdx/package.json:67]: LGTM!

File: packages/vscode-mdx/README.md [hunk 6]
```diff
@@ -16,8 +16,7 @@ You can install this extension from the [Marketplace](https://marketplace.visual
 
 This extension provides the following settings:
 
-*   `mdx.experimentalLanguageServer`: Enable experimental IntelliSense support
-    for MDX files.  (`boolean`, default: false)
+*   `mdx.server.enable`: Enable the MDX language server. (`boolean`, default: true)
```
Comment (remcohaszing) [inline packages/vscode-mdx/README.md:19]:
```suggestion
*   `mdx.server.enable`: Enable the MDX language server.  (`boolean`, default: true)
```

File: packages/vscode-mdx/README.md [hunk 7]
```diff
@@ -16,8 +16,7 @@ You can install this extension from the [Marketplace](https://marketplace.visual
 
 This extension provides the following settings:
 
-*   `mdx.experimentalLanguageServer`: Enable experimental IntelliSense support
-    for MDX files.  (`boolean`, default: false)
+*   `mdx.server.enable`: Enable the MDX language server.  (`boolean`, default: true)
```
Comment (remcohaszing) [inline packages/vscode-mdx/README.md:19]:
```suggestion
*   `mdx.server.enable`: Enable the MDX language server.
    (`boolean`, default: true)
```

**Comment by remcohaszing** [comment]
Thanks! :heart:

**Pull request merged**